### PR TITLE
[MNOE-483] Several Fixes

### PR DESCRIPTION
**Pull request by x4d3**
- Disable Link to Clients Management for Admin
- Flag SubTenant Feature
- Add Staff local list directive

### CODE REVIEW

File: src/app/index.default-config.coffee [hunk 1]
```diff
@@ -6,3 +6,4 @@ angular.module('mnoEnterprise.defaultConfiguration', [])
   .constant('REVIEWS_CONFIG', {enabled: false})
   .constant('ADMIN_PANEL_CONFIG', {})
   .constant('QUESTIONS_CONFIG', {enabled: false})
+  .constant('SUB_TENANT_CONFIG', {enabled: false})
```
Comment (ouranos) [inline src/app/index.default-config.coffee]: Please nest in `ADMIN_PANEL_CONFIG`

File: src/app/index.route.coffee [hunk 2]
```diff
@@ -134,6 +134,7 @@
         label: 'mnoe_admin_panel.dashboard.sub_tenants.title'
       resolve:
         skip: (MnoeCurrentUser) -> MnoeCurrentUser.skipIfNotAdminRole(['admin'])
+        skipCondition: (RoutingHelper, SUB_TENANT_CONFIG) -> RoutingHelper.skipUnlessCondition(SUB_TENANT_CONFIG && SUB_TENANT_CONFIG.enabled)
```
Comment (ouranos) [inline src/app/index.route.coffee]: Wrap the constant in the adminConfig service. This help with providing default value.
`adminConfig.isSubTenantEnabled()`

File: src/app/views/dashboard.controller.coffee [hunk 3]
```diff
@@ -1,27 +1,30 @@
...
-  main.isReviewingEnabled = REVIEWS_CONFIG && REVIEWS_CONFIG.enabled
-  main.areQuestionsEnabled = QUESTIONS_CONFIG && QUESTIONS_CONFIG.enabled
-  main.isFinanceEnabled = MnoeAdminConfig.isFinanceEnabled()
-  main.adminConfig = MnoeAdminConfig
+    main.errorHandler = MnoErrorsHandler
+    main.staffPageAuthorized = STAFF_PAGE_AUTH
+    main.isReviewingEnabled = REVIEWS_CONFIG && REVIEWS_CONFIG.enabled
+    main.areQuestionsEnabled = QUESTIONS_CONFIG && QUESTIONS_CONFIG.enabled
+    main.isFinanceEnabled = MnoeAdminConfig.isFinanceEnabled()
+    main.isSubTenantEnabled = SUB_TENANT_CONFIG && SUB_TENANT_CONFIG.enabled
```
Comment (ouranos) [inline src/app/views/dashboard.controller.coffee]: Wrap the call in `MnoeAdminConfig`

File: src/app/views/dashboard.layout.html [hunk 4]
```diff
@@ -45,7 +45,7 @@
     <li class="sidebar-list" ng-show="main.staffPageAuthorized.indexOf(main.user.admin_role)!=-1" ng-if="::main.adminConfig.isStaffEnabled()" ui-sref-active="active">
       <a ui-sref="dashboard.staffs">{{'mnoe_admin_panel.dashboard.layout.sidebar.staffs' | translate}}<span class="menu-icon fa fa-wrench"></span></a>
     </li>
-    <li class="sidebar-list" ng-show="main.user.admin_role == 'admin'" ng-if="::main.adminConfig.isStaffEnabled()">
+    <li class="sidebar-list" ng-show="main.user.admin_role == 'admin'" ng-if="::main.isSubTenantEnabled">
```
Comment (ouranos) [inline src/app/views/dashboard.layout.html]: This should be an `||` with `adminConfig.isOrganizationManagementEnabled()`?

File: src/app/views/staff/staff.controller.coffee [hunk 5]
```diff
@@ -1,15 +1,17 @@
-@App.controller 'StaffController', ($log, $stateParams, $window, $uibModal, toastr, MnoeCurrentUser, MnoeUsers, MnoeSubTenants, ADMIN_ROLES) ->
+@App.controller 'StaffController', ($log, $stateParams, $window, $uibModal, toastr, MnoConfirm, MnoeCurrentUser, MnoeUsers, MnoeSubTenants, ADMIN_ROLES, SUB_TENANT_CONFIG) ->
   'ngInject'
   vm = this
   vm.isSaving = false
   vm.adminRoles = ADMIN_ROLES
 
   vm.isAdmin = MnoeCurrentUser.user.admin_role == 'admin'
+  vm.subTenantEnabled = SUB_TENANT_CONFIG && SUB_TENANT_CONFIG.enabled
```
Comment (ouranos) [inline src/app/views/staff/staff.controller.coffee]: Wrap this in the `adminConfig` svc

File: src/app/views/staff/staff.controller.coffee [hunk 6]
```diff
@@ -22,13 +24,27 @@
...
+        (response) ->
+          vm.staff = response.data.user
+          vm.staff.admin_role_was = vm.staff.admin_role
+          toastr.success("mnoe_admin_panel.dashboard.staff.update_staff.toastr_success", {extraData: { staff_name: "#{vm.staff.name} #{vm.staff.surname}"}})
+        (error) ->
+          toastr.error("mnoe_admin_panel.dashboard.staff.update_staff.toastr_error", {extraData: { staff_name: "#{vm.staff.name} #{vm.staff.surname}"}})
+          $log.error("An error occurred while updating staff:", error)
+      ).finally(-> vm.isSaving = false)
+
+    if vm.staff.admin_role_was == 'staff' &&  vm.staff.admin_role != 'staff' && vm.staff.client_ids.length
```
Comment (ouranos) [inline src/app/views/staff/staff.controller.coffee]: Probably good to add the same comment you had in mnoe:
> Ask for confirmation if the user is updated to admin or division admin as his clients will be cleared

File: src/app/views/staff/staff.controller.coffee [hunk 7]
```diff
@@ -22,13 +24,27 @@
...
+      modalOptions =
+        closeButtonText: 'mnoe_admin_panel.dashboard.staff.update_staff_role.cancel'
+        actionButtonText: 'mnoe_admin_panel.dashboard.staff.update_staff_role.action'
+        headerText: 'mnoe_admin_panel.dashboard.staff.update_staff_role.proceed'
+        bodyText: 'mnoe_admin_panel.dashboard.staff.update_staff_role.perform'
+      MnoConfirm.showModal(modalOptions).then(updateStaffAction).finally(-> vm.isSaving = false)
+    else
+      updateStaffAction()
```
Comment (alexnoox) [inline src/app/views/staff/staff.controller.coffee]: `updateStaffAction()` is only used here, maybe you should implement it directly here.

File: src/app/views/sub-tenant/sub-tenant.html [hunk 8]
```diff
@@ -22,13 +22,13 @@ <h1 translate><i class="fa fa-institution"></i>mnoe_admin_panel.dashboard.sub_te
 
     <div class="bs-row">
       <div class="col-sm-6">
-        <h2><i class="fa fa-users" translate></i> Clients</h2>
+        <h2 translate><i class="fa fa-users" translate></i>mnoe_admin_panel.dashboard.sub_tenant.clients</h2>
```
Comment (alexnoox) [inline src/app/views/sub-tenant/sub-tenant.html]: You can remove some `translate` directives in this file

File: src/app/views/staff/staff.controller.coffee (hunk 7 above)
Comment (x4d3) [inline src/app/views/staff/staff.controller.coffee]: It is also used in the then of the MnoConfirm

File: src/app/views/dashboard.layout.html (hunk 4 above)
Comment (x4d3) [inline src/app/views/dashboard.layout.html]: you mean isStaffEnabled ?

**Comment by x4d3** [comment]
@alexnoox @ouranos  I've applied your change requests.

File: src/app/views/dashboard.layout.html (hunk 4 above)
Comment (ouranos) [inline src/app/views/dashboard.layout.html]: Hmm indeed. I can't remember why I wrote that...

**Pull request merged**